kubectl-cloudflow/cmd/deploy.go

```diff
@@ -153,14 +153,8 @@ func (opts *deployOptions) deployImpl(cmd *cobra.Command, args []string) {
 		}
 	}
 
-	// Get the Cloudflow operator ownerReference
-	ownerReference := version.GetOwnerReferenceForCloudflowOperator()
-
-	serviceAccount := newCloudflowServiceAccountWithImagePullSecrets(namespace, ownerReference)
-
-	if _, err := createOrUpdateServiceAccount(k8sClient, namespace, serviceAccount); err != nil {
-		util.LogAndExit("%s", err)
-	}
+	// Delay the creation of the service account for after the ownerReferences has been generated.
+	serviceAccount := newCloudflowServiceAccountWithImagePullSecrets(namespace)
```

Comment:
The creation of `serviceAccount` and `streamletNameSecretMap` can be moved later in the code, where they are actually used.

Reply:
Fixed.